src/dmd/initsem.d

```diff
@@ -721,17 +721,17 @@ extern(C++) Initializer initializerSemantic(Initializer init, Scope* sc, ref Typ
         {
             //printf(" type %s i %d dim %d dil.length = %d\n", t.toChars(), cast(int)i, cast(int)dim, cast(int)dil.length);
             auto tn = t.nextOf().toBasetype();
-            if (auto tna = tn.isTypeDArray())
+            auto tnsa = tn.isTypeSArray();
+            if (tnsa && tnsa.isIncomplete())
```

Comment:
Shouldn't this condition also check that that we are in cfile scope?

Reply:
That's implicit in it being in CInitializer code.

Reply:
Oh, I missed the fact that we are in `visitC`.